code/modules/recycling/sortingmachinery.dm

```diff
@@ -1,35 +1,39 @@
-/obj/structure/big_delivery
-	name = "large parcel"
-	desc = "A large delivery parcel."
+/obj/item/delivery
 	icon = 'icons/obj/storage.dmi'
-	icon_state = "deliverycloset"
-	density = TRUE
-	mouse_drag_pointer = MOUSE_ACTIVE_POINTER
-	var/giftwrapped = FALSE
+	var/giftwrapped = 0
 	var/sort_tag = 0
 	var/obj/item/paper/note
 	var/obj/item/barcode/sticker
 
-/obj/structure/big_delivery/Initialize(mapload)
+/obj/item/delivery/Initialize(mapload)
 	. = ..()
 	RegisterSignal(src, COMSIG_MOVABLE_DISPOSING, .proc/disposal_handling)
 
-/obj/structure/big_delivery/interact(mob/user)
+/**
+ * Initial check if manually unwrapping
+ */
+/obj/item/delivery/proc/attempt_pre_unwrap_contents(mob/user)
 	to_chat(user, span_notice("You start to unwrap the package..."))
-	if(!do_after(user, 15, target = user))
+	return do_after(user, 15, target = user)
+
+/**
+ * Signals for unwrapping.
+ */
+/obj/item/delivery/proc/unwrap_contents()
+	if(!sticker)
 		return
+	for(var/obj/I in src.get_all_contents())
```

Comment:
no reason to use explicit src and use a real variable name for the iterator

Reply:
also why get_all_contents() instead of just contents?

Reply:
This line was actually from the old code. I can improve the var name. It's not clear to me if it's intentional that the signal is called recursively or not. Currently the only consumer of that signal is taped on price tags that are direct children of the parcel. I can make it just use direct contents for no change in behavior. Still not sure what the intended behavior of the signal was.

Reply:
All instances of those problems fixed as far as I can see.